tests/test_external_search.py

```diff
@@ -396,490 +453,275 @@ def test_query_works(self):
+        fantasy, ignore = Genre.lookup(self._db, "Fantasy")
+        biography_filter = Filter(genre_restriction_sets=[[biography]])
+        fantasy_filter = Filter(genre_restriction_sets=[[fantasy]])
+        both = Filter(genre_restriction_sets=[[fantasy, biography]])
 
-        results = query("lincoln", None, None, None, None, None, both_lane.genre_ids)
-        hits = results["hits"]["hits"]
-        eq_(2, len(hits))
+        expect(self.lincoln, "lincoln", biography_filter)
+        expect(self.lincoln_vampire, "lincoln", fantasy_filter)
+        expect([self.lincoln, self.lincoln_vampire], "lincoln", both,
+               ordered=False)
 
         # Filters on list membership.
+
         # This ignores 'Abraham Lincoln, Vampire Hunter' because that
-        # book isn't on the list.
-        results = query("lincoln", None, None, None, None, None, None, on_any_of_these_lists=[self.presidential.id])
-        hits = results['hits']['hits']
-        eq_(1, len(hits))
-        eq_(unicode(self.lincoln.id), hits[0]["_id"])
+        # book isn't on the self.presidential list.
+        on_presidential_list = Filter(
+            customlist_restriction_sets=[[self.presidential]]
+        )
+        expect(self.lincoln, "lincoln", on_presidential_list)
 
         # This filters everything, since the query is restricted to
         # an empty set of lists.
-        results = query("lincoln", None, None, None, None, None, None, on_any_of_these_lists=[])
-        hits = results['hits']['hits']
-        eq_(0, len(hits))
-
-        # This query does not match anything because the book in
-        # question is not in a collection associated with the default
-        # library.
-        results = query("a tiny book", None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(0, len(hits))
-
-        # If we don't pass in a library to query_works, the entire index is
-        # searched and we can see everything regardless of which collection
-        # it's in.
-        results = self.search.query_works(
-            None, "book", None, None, None, None, None, None, None
-        )
-        hits = results["hits"]["hits"]
-        eq_(1, len(hits))
-        results = self.search.query_works(
-            None, "moby dick", None, None, None, None, None, None, None
-        )
-        hits = results["hits"]["hits"]
-        eq_(2, len(hits))
+        expect([], "lincoln", Filter(customlist_restriction_sets=[[]]))
 
-        #
-        # Test searching across collections.
-        #
 
-        # If we add the missing collection to the default library, "A
-        # Tiny Book" starts showing up in searches against that
-        # library.
-        self._default_library.collections.append(self.tiny_collection)
-        results = query("a tiny book", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(1, len(hits))
```

Comment:
A lot of this test code could be simplified because we "search the current library" by setting an appropriate value for `Filter.collection_ids` when creating the `Filter` object. It's no longer any special code run during the search itself.